### PR TITLE
Pre release fixing

### PR DESCRIPTION
**Pull request by haaakon**

### CODE REVIEW

File: Digipost/APIClient.swift [hunk 1]
```diff
@@ -445,6 +452,8 @@ class APIClient : NSObject, NSURLSessionTaskDelegate, NSURLSessionDelegate, NSUR
                 }, failure: { (error) -> Void in
                     if error.code == Int(SHCOAuthErrorCode.InvalidRefreshTokenResponse.rawValue) {
                         self.deleteRefreshTokensAndLogoutUser()
+                    } else {
+                        Logger.dpostLogError("Deleting oauth tokens and logout user. Not invalid refresh token.  error: \(error)")
```
Comment (lillesand) [inline Digipost/APIClient.swift]: Correct this error message

Comment (lillesand) [inline Digipost/APIClient.swift]: Don't log the full error object, be more specific, so we don't accidentally log something we don't want to

File: Digipost/POSDocumentsViewController.m [hunk 2]
```diff
@@ -521,7 +521,7 @@ - (void)updateContentsFromServerUserInitiatedRequest:(NSNumber *)userDidInititat
         
         [self showTableViewBackgroundView:([self numberOfRows] == 0)];
         if ([userDidInititateRequest boolValue]){
-            [UIAlertController presentAlertControllerWithAPIError:error presentingViewController:self];
+            [UIAlertController presentAlertControllerWithAPIError:error presentingViewController:self didTapOkClosure:nil];
```
Comment (lillesand) [inline Digipost/POSDocumentsViewController.m]: Consider using default value for didTapOkClosure

File: Digipost/POSLetterViewController.m [hunk 3]
```diff
@@ -899,9 +901,16 @@ - (void)removeUnlockViewIfPresent
...
     }
 }
+
 - (void)unloadContent
 {
     [[POSFileManager sharedFileManager] removeAllDecryptedFiles];
+    if (self.attachment) {
+        NSString *currentScope = [OAuthToken oAuthScopeForAuthenticationLevel:self.attachment.authenticationLevel];
+        if (currentScope != kOauth2ScopeFull) {
+            [self.attachment deleteEncryptedFileIfExisting];
```
Comment (lillesand) [inline Digipost/POSLetterViewController.m:908]: Add a comment on why we do this. The rule is not obvious :wink:

File: Digipost/UIAlertController+APIError.swift [hunk 4]
```diff
@@ -10,35 +10,22 @@ import UIKit
...
+
             alertController.addAction(okAction)
-            presentingViewController.presentViewController(alertController, animated: true, completion: { () -> Void in
-                
-            })
-            
+
+            if presentingViewController.navigationController?.topViewController == presentingViewController && presentingViewController.presentedViewController == nil {
+                presentingViewController.presentViewController(alertController, animated: true, completion:nil)
+                var descriptionWithNoForbiddenCharacters = "\(presentingViewController)".stringByReplacingOccurrencesOfString("<", withString: "")
```
Comment (lillesand) [inline Digipost/UIAlertController+APIError.swift]: Comment why we do this

**Comment by lillesand** [comment]
Good work! Review comments and merge when ready.

File: Digipost/APIClient.swift (hunk 1 above)
Comment (haaakon) [inline Digipost/APIClient.swift]: I did a small rewrite. I either want to logout user if refreshtoken is out, or i want to to let the failure block handle the error. If the error is shown to user, it's logged by Logger there instead. 

commit: [0edc84](https://github.com/digipost/ios/commit/0edc84112b8ada730a3ffb62f5aef600b56e9a65)

File: Digipost/POSDocumentsViewController.m (hunk 2 above)
Comment (haaakon) [inline Digipost/POSDocumentsViewController.m]: I actually found a pretty insane bug in the swift compiler/translator to objective-C when trying this.
 Sadly it won't compile at present Xcode 6.3 with Swift 1.2 :sob:

Comment (haaakon) [inline Digipost/POSDocumentsViewController.m]: I'll probably report it to Apple's radar bug system, i can link to that when added

Comment (haaakon) [inline Digipost/POSDocumentsViewController.m]: Allright, found a workaround! commit: 500d376fabfbf9f509ccfe495a75cd1ffffedfde

File: Digipost/POSLetterViewController.m (hunk 3 above)
Comment (haaakon) [inline Digipost/POSLetterViewController.m:908]: commit: 6712b5f57bcd96f9bbfcf19736afb2de893bf321

File: Digipost/UIAlertController+APIError.swift (hunk 4 above)
Comment (haaakon) [inline Digipost/UIAlertController+APIError.swift]: removed logging which view that got the alert, i don't think the information is that valuable. It's easily understandable from the alert code also.
commit: 802fddbe390b5b38c97c6d731c00e182f4abbd35

**Pull request merged**